### PR TITLE
Hooks pro pods add image pull secrets

### PR DESCRIPTION
**Pull request by kmadel**
Although the vCluster vcluster.yaml configuration provides the workloadServiceAccount that allows including `imagePullSecrets`, it is limited as that `serviceAccount` is superseded when using `sync.toHost.pods.serviceAccounts.enabled: true`.

This allows setting `imagePullSecrets` at the pod level instead of via a `serviceAccount` - although it does rely on a Pro feature.

### CODE REVIEW

**Comment by Piotr1215** [comment]
Could you please [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so I can help you with `vale`?

**Comment by kmadel** [comment]
The Vale issue is related to using "Hooks (Pro)". I believe we use this formatting in multiple places. If we are going to change it to satisfy Vale, what should it be changed to? "Hooks (pro)"? Also, can't add comment in source as this Vale issue is for something that was not changed in the PR.

File: vcluster/configure/vcluster-yaml/experimental/generic-sync.mdx
Comment (Piotr1215) [inline vcluster/configure/vcluster-yaml/experimental/generic-sync.mdx:1]: You can switch off vale rules by adding comment above and below, like so:


## Hooks (Pro)


I will add the (Pro) to exceptions.

Comment (Piotr1215) [reply vcluster/configure/vcluster-yaml/experimental/generic-sync.mdx:1]: PR with the exception: https://github.com/loft-sh/vcluster-docs/pull/351

Once it's merged, and you rebase your branch off of main, the warning should go away

**Comment by Piotr1215** [comment]
@kmadel you need to either rebase or trigger a new commit on your branch, something like:

```git
git commit --allow-empty -m "trigger CI after merge from main"
git push
```

**Review by Piotr1215 — APPROVED** [review]

**Pull request merged**
